### PR TITLE
Fix early exit for OR expression

### PR DESCRIPTION
**Pull request by Josipmrden**
Fix or expression not making an early exit if first branch is evaluated to true.

[master < Task] PR
- [X] Check, and update documentation if necessary
- [X] Provide the full content or a guide for the final git message

To keep docs changelog up to date, one more thing to do:
- [X] Write a release note here, including added/changed clauses
- [X] Tag someone from docs team in the comments

Release note: "Added an early exit when doing OR filter expressions if the first expression evaluates to `true`"
@kgolubic

### CODE REVIEW

File: src/query/interpret/eval.hpp [hunk 1]
```diff
@@ -261,6 +260,20 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
...
 
+  TypedValue Visit(OrOperator &op) override {
+    auto value1 = op.expression1_->Accept(*this);
+    if (value1.IsBool() && value1.ValueBool()) {
+      // If first expression is true, don't evaluate the second one.
+      return value1;
+    }
+    auto value2 = op.expression2_->Accept(*this);
+    try {
+      return value1 || value2;
```
Comment (as51340) [inline src/query/interpret/eval.hpp:271]: if 
```suggestion
      if (value2.IsBool()) return value2;
      return false;
```

Comment (as51340) [reply src/query/interpret/eval.hpp:271]: if we came to this point, value1 is not bool or false so I think correct thing is just to return value2 ?

**Review by as51340 — CHANGES_REQUESTED** [review]

**Review by as51340 — APPROVED** [review]
:muscle:

File: src/query/interpret/eval.hpp [hunk 2]
```diff
@@ -261,6 +260,20 @@ class ExpressionEvaluator : public ExpressionVisitor<TypedValue> {
     }
   }
 
+  TypedValue Visit(OrOperator &op) override {
+    auto value1 = op.expression1_->Accept(*this);
+    if (value1.IsBool() && value1.ValueBool()) {
+      // If first expression is true, don't evaluate the second one.
```
Comment (antoniofilipovic) [inline src/query/interpret/eval.hpp:266]: No need for comment

**Review by antoniofilipovic — APPROVED** [review]
LGTM, left comment tiny

**Comment by kgolubic** [comment]
Release note entry:

> Added an early exit when doing `OR` filter expressions if the first expression evaluates to `true`.

**Pull request merged**